app/src/org/commcare/activities/connect/viewmodel/PersonalIdSessionDataViewModel.java

```diff
@@ -2,24 +2,30 @@
 
 import org.commcare.android.database.connect.models.PersonalIdSessionData;
 
+import androidx.lifecycle.SavedStateHandle;
 import androidx.lifecycle.ViewModel;
 
 public class PersonalIdSessionDataViewModel extends ViewModel {
-    private PersonalIdSessionData personalIdSessionData;
+    private static final String PERSONAL_ID_SESSION_DATA_KEY = "PERSONAL_ID_SESSION_DATA_KEY";
+    private final SavedStateHandle savedStateHandle;
+
+    public PersonalIdSessionDataViewModel(SavedStateHandle savedStateHandle) {
+        this.savedStateHandle = savedStateHandle;
+    }
```

Comment:
Does this constructor gets called by itself when activity gets re-created, resulting in re-creation of the `PersonalIdSessionDataViewModel` as it was saved ?

Reply:
Yes, when a `savedStateHandle` is in the viewmodel's constructor, the Android framework provides it to the viewmodel automatically, and the `savedStateHandle` will contain any saved data when the state is restored